hazelcast/src/main/java/com/hazelcast/datastore/impl/ExternalDataStoreServiceImpl.java

```diff
@@ -61,11 +64,23 @@ private ExternalDataStoreFactory<?> createFactory(ExternalDataStoreConfig config
     }
 
     @Override
-    public ExternalDataStoreFactory<?> getExternalDataStoreFactory(String name) {
+    public <DS> ExternalDataStoreFactory<DS> getExternalDataStoreFactory(String name) {
         ExternalDataStoreConfig externalDataStoreConfig = node.getConfig().getExternalDataStoreConfigs().get(name);
         if (externalDataStoreConfig == null) {
             throw new HazelcastException("External data store factory '" + name + "' not found");
         }
-        return dataStoreFactories.computeIfAbsent(name, n -> createFactory(externalDataStoreConfig));
+        return (ExternalDataStoreFactory<DS>) dataStoreFactories
+                .computeIfAbsent(name, n -> createFactory(externalDataStoreConfig));
+    }
+
+    @Override
+    public void close() {
```

Comment:
minor: maybe having a finest level log to print start of closing data-store-factories can be useful for diagnostic purposes.